content/en/blog/2021/proxyless-grpc/index.md

```diff
@@ -0,0 +1,400 @@
+These APIs aim to become a [universal data-plane API](https://blog.envoyproxy.io/the-universal-data-plane-api-d15cec7a?gi=64aa2eea0283).
+The gRPC project has started to implement the xDS APIs, which means you can manage gRPC workloads without having to deploy
+an Envoy sidecar along with them. You can learn more about the integration in a [KubeCon EU 2021 talk from Megan Yahya](https://www.youtube.com/watch?v=cGJXkZ7jiDk).
+
+Istio 1.11 adds experimental support for adding gRPC services directly to the mesh. We support basic service
+discovery, some VirtualService based traffic policy, and mutual TLS.
+
+## Supported Features
+
+The current implementation of the xDS APIs within gRPC is limited in some areas compared to Envoy. The following
+features should work, although this is not an exhaustive list and other features may have partial functionality:
+
+* Basic service discovery. Your gRPC service can reach other pods and virtual machines registered in the mesh.
+* [`DestinationRule`](/docs/reference/config/networking/destination-rule/):
+    * Subsets: Your gRPC service can split traffic based on label selectors to different groups of instances.
+    * The only Istio `loadBalancer` currently supported is `ROUND_ROBIN`, `consistentHash` will be added in the future.
+    * `tls` settings are restricted to `DISABLE` or `ISTIO_MUTUAL`. Other modes will be treated as `DISABLE`.
+* [`VirtualService`](/docs/reference/config/networking/virtual-service/):
+    * Header match and URI match in the format `/ServiceName/RPCName`
+    * Override destination host and subset.
+    * Weighted traffic shifting.
+* [`PeerAuthentication`](/docs/reference/config/security/peer_authentication/):
+    * Only `DISABLE` and `STRICT` are supported. Other modes will be treated as `DSIABLE`.
+    * Support for auto-mTLS may exist in a future release.
+
+Other features including faults, retries, timeouts, mirroring and rewrite rules may be supported in a future release.
+Some of these features are awaiting implementation in gRPC, and others require work in Istio to support. The status
+of xDS features in gRPC can be found [here](https://github.com/grpc/grpc/blob/master/doc/grpc_xds_features.md). The
+status of Istio's support will exist in future official docs.
+
+{{< warning >}}
+This is feature is [experimental](/docs/releases/feature-stages/). Standard Istio features will become supported
+over time along with improvements to the overall design.
+{{< /warning >}}
+
+## Architecture Overview
+
+{{< image width="80%" link="./architecture.svg" caption="Diagram of how gRPC services communicate with the istiod" >}}
+
+Although this doesn't use a proxy for data plane communication, it still requires an agent for initialization and
+communication with the control-plane. First, the agent generates a [bootstrap file](https://github.com/grpc/proposal/blob/master/A27-xds-global-load-balancing.md#xdsclient-and-bootstrap-file).
```

Comment:
Sorry, it is not clear to me how the agent generates this file. Could you expand it a bit?

Reply:
Hopefully I made it clearer – it generates this on startup using various Istio configurations to set the node metadata and discovery address.